src/lib/optimizer/join_ordering/join_edge.hpp

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include <iostream>
+#include <ostream>
+
+#include "boost/dynamic_bitset.hpp"
+
+namespace opossum {
+
+class AbstractJoinPlanPredicate;
+
+/**
+ * Represents a (hyper)edge in a JoinGraph. Each predicate must operate exactly on the vertices in vertex_set
```

Comment:
Added the comment "That is, each predicate must reference columns from all vertices in vertex_set and no columns from vertices not in vertex_set. If the predicate wouldn't, then it would belong to another edge."